.gitlab-ci.yml

```diff
@@ -1,8 +1,76 @@
 stages:
   - test
   - benchmark
+  - execute
   - upload
 
+
+# job_juwels_compute:
+#   stage: execute
+#   variables:
+#     SCHEDULER_PARAMETERS: '--account=cstma --nodes=1 --partition=devel'
+#   tags:
+#     - juwels
+#     - jacamar
+#     - compute
+#     - slurm
+#   artifacts:
+#     paths:
+#       - test.file
+#   script:
+#     - echo $SYSTEMNAME
+#     - touch test.file
+#   after_script:
+#     - hostname
+#     - id
+
+
+variables:
+  JUWELS_ACCOUNT: "cstma"
+
+
+test_JUWELS:
+  stage: benchmark
+  rules:
+    - if: $CI_COMMIT_MESSAGE !~ /.*\[CI-no-benchmarks\]/
+  tags:
+    - jacamar
+    - juwels
+    - login
+    - shell
+  artifacts:
+    when: always
+    paths:
+      - benchmarks
+      - sbatch.err
+      - sbatch.out
+  before_script:
+    - mkdir -p benchmarks
+    # load the latest Python module (currently 3.11)
+    - module --force purge
+    - module load Stages/2024
+    - module load GCC
+    - module load OpenMPI
+    - module load FFTW
+    - module load mpi4py
+    - module load SciPy-Stack
+    - module load CuPy
+    - pwd
+    - ls -lah
+    - pip install -e .
```

Comment:
Do you need to `pip install` in the CI? User installation with pip is persistent, afaik. Is the gitlab not associated with any user?
I would prefer to put the `module load` commands in the job scripts. But if you need to repeat the `pip install`s every time, it's better here.

Reply:
I'm not sure, if we need `pip install` each time. The jobs are run as a predefined user (the one who triggers the CI in Gitlab). As mirroring is done with a personal access token, that user is impersonated and the CI will always run as this specific user.

Regarding `module load`: We can put them in the script. I find it easier/nicer if the scripts are as short as possible. Therefore, I moved the `module load` into the YAML-file. Furthermore, these steps (as module load or pip install) are executed on a login-node. The content of the sh-file is executed on a compute-node. In term of quota, it is "cheaper" if we don't spend compute-time for module load (although it does not take too long).

 If you want to, feel free to move the pip install and module load into the script. If I shall do that, feel free to ping me!